### PR TITLE
docs: Document Create Huggingface Dataset Pipeline

### PR DESCRIPTION
**Pull request by kriangkraitan**
## Why this PR
Why we need this PR?

## Changes
- Write some changes here

## Related Issues
Close #

## Checklist
- [ ] PR should be in the [Naming convention](../../docs/PR_NAMING.md)
- [ ] Assign yourself in to Assigneees
- [ ] Tag related issues
- [ ] Constants name should be ALL_CAPITAL, function name should be snake_case, and class name should be CamelCase
- [ ] complex function/algorithm should have [Docstring](https://peps.python.org/pep-0257/)
- [ ] 1 PR should not have more than 200 lines changes (Exception for test files). If more than that please open multiple PRs
- [ ] At least PR reviewer must come from the task's team (model, eval, data)

### CODE REVIEW

File: src/data/scripts/huggingface_create/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+## Dependencies
+Before using this script, make sure you have the following dependencies installed:
+
+datasets
+pandas
+tqdm
+You can typically install these dependencies using pip:
+
+```bash
+pip install datasets pandas tqdm
```
Comment (new5558) [inline src/data/scripts/huggingface_create/README.md:16]: Can you lock the version for the library of this script? In the future, if datasets or pandas are updated to a breaking change version (X.x.x) and we don't have the correct version mentioned in the readme, this script will become obsolete.

File: src/data/scripts/huggingface_create/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+The input data should be in JSONL (JSON Lines) format, where each line represents a JSON object. Each JSON object represents a data instance with one or more fields.
+
+The script assumes that all JSON objects within the JSONL files have the same set of keys. If some keys are missing in certain objects, they will be added with a default value of None.
+
+## Output Data Format
+The script converts the input JSONL data into a Hugging Face dataset format. The resulting dataset is saved as a directory, and it contains two splits:
+
+train: The training split.
+eval: The evaluation split.
+Each split is stored as a set of Parquet files and can be loaded and used with Hugging Face Datasets library.
```
Comment (new5558) [inline src/data/scripts/huggingface_create/README.md:40]: I thought the split will be stored in arrow file, not parquet. Can you confirm this information?

File: src/data/scripts/merge_jsonl/README.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# JSONL File Merger
+
+This Python script merges JSONL (JSON Lines) files from a specified folder into a single output file.
```
Comment (new5558) [inline src/data/scripts/merge_jsonl/README.md:3]: Where is `## Dependencies` ex. `pip install ...`?

File: src/data/scripts/merge_jsonl/README.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+
+## Usage
+To merge JSONL files in a folder and save the merged data to an output file, In the SLURM script, modify the last line to specify the input folder containing the JSONL files and the output file path:
+
+```bash
+python merge_jsonl_files.py folder_path output_file
```
Comment (new5558) [inline src/data/scripts/merge_jsonl/README.md:9]:
```suggestion
python merge_jsonl_files.py <folder_path> <output_file.jsonl>
```

File: src/data/scripts/huggingface_create/README.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+```bash
+pip install datasets pandas tqdm
+```
+
+## How to Run
+You can run this script from the command line with the following command:
+
+```bash
+python script.py --train_path path_to_train_data --eval_path path_to_eval_data --output_path output_dataset_directory
```
Comment (new5558) [inline src/data/scripts/huggingface_create/README.md:23]:
```suggestion
python script.py --train_path <rain_data_directory> --eval_path <eval_data_directory> --output_path <output_dataset_directory>
```

File: src/model/scripts/finetune_llama2/OpenThaiGPT_FineTune_Llama.ipynb [hunk 6]
```diff
@@ -0,0 +1,1474 @@
+{
```
Comment (new5558) [inline src/model/scripts/finetune_llama2/OpenThaiGPT_FineTune_Llama.ipynb:1]: Can you put this into folder `src/model/notebook/finetune_llama2`. If the folder did not exist, please create it.

File: src/data/scripts/split_data/README.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
+# JSONL File Splitter
+This Python script is designed to split a large JSONL (JSON Lines) file into three separate sets: train, test, and validation. It's a useful tool for preparing data for training, ensuring that you have distinct datasets for training, testing, and validation.
+
+## Usage
+In the SLURM script, modify the last line to specify the input JSONL file path and the desired test and validation set sizes:
+```bash
+python split_jsonl.py /path/to/your/jsonl/file.jsonl --test_size 0.001 --validation_size 0.001
```
Comment (new5558) [inline src/data/scripts/split_data/README.md:7]:
```suggestion
python split_jsonl.py <file.jsonl> --test_size 0.001 --validation_size 0.001
```

File: src/data/scripts/split_data/README.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+```
+Replace /path/to/your/jsonl/file.jsonl with the path to your JSONL file.
```
Comment (new5558) [inline src/data/scripts/split_data/README.md:9]:
```suggestion
Replace <file.jsonl> with the path to your JSONL file.
```

File: src/data/scripts/huggingface_create/README.md (hunk 2 above)
Comment (kriangkraitan) [reply src/data/scripts/huggingface_create/README.md:40]: Refer to the information in this link. https://huggingface.co/docs/datasets/about_arrow#memorymapping I think the split will be stored in arrow file.

File: src/data/scripts/huggingface_create/README.md (hunk 1 above)
Comment (kriangkraitan) [reply src/data/scripts/huggingface_create/README.md]: done

File: src/data/scripts/merge_jsonl/README.md (hunk 3 above)
Comment (kriangkraitan) [reply src/data/scripts/merge_jsonl/README.md:3]: done

File: src/model/scripts/finetune_llama2/OpenThaiGPT_FineTune_Llama.ipynb (hunk 6 above)
Comment (kriangkraitan) [reply src/model/scripts/finetune_llama2/OpenThaiGPT_FineTune_Llama.ipynb]: done

File: src/data/scripts/huggingface_create/README.md [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+## Dependencies
+Before using this script, make sure you have the following dependencies installed:
+
+datasets>=2.12.0
+pandas>=2.0.2
+tqdm>=4.65.0
+You can typically install these dependencies using pip:
+
+```bash
+pip install datasets>=2.12.0 pandas>=2.0.2 tqdm>=4.65.0
```
Comment (new5558) [inline src/data/scripts/huggingface_create/README.md:16]: `>=` means when datasets 3.0.0 is released, this code will have a high chance of breaking. this symbol `X.x.x` , the first X version usually means "breaking changes". "breaking changes" represent library API change that the application code may need modification to be able to have the same functionality.

File: src/data/scripts/merge_jsonl/README.md [hunk 10]
```diff
@@ -0,0 +1,21 @@
+# JSONL File Merger
+
+This Python script merges JSONL (JSON Lines) files from a specified folder into a single output file.
+
+## Dependencies
+Before using this script, make sure you have the following dependencies installed:
+
+jsonlines>=3.1.0
```
Comment (new5558) [inline src/data/scripts/merge_jsonl/README.md:8]: `>=` ??

File: src/data/scripts/merge_jsonl/README.md [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+You can typically install these dependencies using pip:
+
+```bash
+pip install jsonlines>=3.1.0
```
Comment (new5558) [inline src/data/scripts/merge_jsonl/README.md:12]: `>=` ??

File: src/model/notebook/finetune_llama2/OpenThaiGPT_FineTune_Llama.ipynb [hunk 12]
```diff
@@ -0,0 +1,1474 @@
+{
```
Comment (new5558) [inline src/model/notebook/finetune_llama2/OpenThaiGPT_FineTune_Llama.ipynb:1]: I will not go through reviewing this notebook code but please make sure that

1. Please add readme for this notebook (just quick and simple readme is enough)
2. If this notebook is Colab exported
    2.1.  `pip install ...` should be presented inside
    2.2.  please add "Open in Colab" link in readme. example: https://colab.research.google.com/github/googlecolab/colabtools/blob/master/notebooks/colab-github-demo.ipynb

3. If this notebook is intended to run on Lanta
3.1 please make sure this code can be run by just `pip install -r ./src/model` if not please describe how to install depedencies

**Review by new5558 — APPROVED** [review]

File: src/data/scripts/huggingface_create/README.md (hunk 9 above)
Comment (kriangkraitan) [reply src/data/scripts/huggingface_create/README.md]: done

File: src/data/scripts/huggingface_create/README.md [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+```bash
+python script.py --train_path <rain_data_directory> --eval_path <eval_data_directory> --output_path <output_dataset_directory>
+```
+
+train_path: Path to the directory containing the training JSONL files.
+eval_path: Path to the directory containing the evaluation JSONL files.
+output_path: Path to the directory where the resulting dataset will be saved.
+
+## Input Data Format
+The input data should be in JSONL (JSON Lines) format, where each line represents a JSON object. Each JSON object represents a data instance with one or more fields.
```
Comment (kwankoravich) [inline src/data/scripts/huggingface_create/README.md:31]: more files?

**Review by kwankoravich — APPROVED** [review]

**Pull request merged**